### PR TITLE
Added rever

### PR DESCRIPTION
**Pull request by melissawm**
Preparing for release.

### CODE REVIEW

**Comment by asmeurer** [comment]
I don't see anything that's missing. Maybe @scopatz can give it a quick look too.

File: rever.xsh [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+def annotated_tag():
+    # https://github.com/regro/rever/issues/212
+    git tag -a -m "$GITHUB_REPO $VERSION release" $VERSION
+
+$ACTIVITIES = [
+            'run_tests',
+            'build_docs',
+            'annotated_tag', # Creates a tag for the new version number
+            'pypi',  # Sends the package to pypi
+            'push_tag',  # Pushes the tag up to the $TAG_REMOTE
```
Comment (scopatz) [inline rever.xsh:33]: Shouldn't pypi come after push_tag?

File: rever.xsh [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+        make html
+        cd ..
+
+@activity
+def annotated_tag():
+    # https://github.com/regro/rever/issues/212
+    git tag -a -m "$GITHUB_REPO $VERSION release" $VERSION
+
+$ACTIVITIES = [
+            'run_tests',
```
Comment (scopatz) [inline rever.xsh:29]: We should probably have an "authors" activity and a "changelog" activity (unless we are using the same webservice sympy is for changelogs)

Comment (asmeurer) [reply rever.xsh:29]: For ndindex at least I just write the changelog by hand. https://quansight.github.io/ndindex/changelog.html.

The SymPy bot currently isn't easily reusable. That's something I could potentially work on, though I might need some advice from some other folks here on how to best do it.

**Comment by scopatz** [comment]
@melissawm - you probably want to rerun the rever setup tool to init authors and changelog. Also, you can see what I did in conda-docker to use markdown for both of these activities https://github.com/regro/conda-docker/blob/master/rever.xsh#L14

File: AUTHORS.md [hunk 3]
```diff
@@ -0,0 +1,9 @@
+All of the people who have made at least one contribution to versioned-hdf5.
+Authors are sorted by number of commits.
```
Comment (asmeurer) [inline AUTHORS.md:2]: Is this file generated automatically?

Comment (asmeurer) [reply AUTHORS.md:2]: I would sort this by last name.

File: .authors.yml [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+  email: bessen@deshaw.com
+  num_commits: 3
+  first_commit: 2020-04-03 15:38:29
+  github: ArvidJB
+- name: Melissa Weber Mendonça
+  email: melissawm@gmail.com
+  aliases:
+  - Melissa Weber Mendonca
+  - MelissaWM
+  - melissawm
```
Comment (asmeurer) [inline .authors.yml:11]: Shouldn't these be managed by a .mailmap file?

File: CHANGELOG.md [hunk 5]
```diff
@@ -0,0 +1,5 @@
+=========================
```
Comment (asmeurer) [inline CHANGELOG.md:1]: This should go in the docs.

**Comment by asmeurer** [comment]
Let's also make sure that all the extra files are included in the tarball with the MANIFEST.in.

File: .authors.yml (hunk 4 above)
Comment (scopatz) [reply .authors.yml:11]: No, this produces the mailmap file

File: AUTHORS.md (hunk 3 above)
Comment (scopatz) [reply AUTHORS.md:2]: Yes, this is generated automatically.  I am not sure there is a sort-by-lastname option

Comment (scopatz) [reply AUTHORS.md:2]: That would be an awesome addition to rever though, if you want to spend some time on it either of you

File: .authors.yml (hunk 4 above)
Comment (asmeurer) [reply .authors.yml:11]: Oh cool. We should generate that file and check it into the repo.

Comment (scopatz) [reply .authors.yml:11]: I think it generates it and checks it in on release

Comment (asmeurer) [reply .authors.yml:11]: Oh it actually commits it on release? That's interesting.

File: rever.xsh [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+from rever.conda import run_in_conda_env
+
+$PROJECT = 'versioned-hdf5'
+
+@activity
+def run_tests():
+    # Don't use the built-in pytest action because that uses Docker, which is
+    # overkill and requires installing Docker
+    with run_in_conda_env(['python=3.8', 'pytest', 'numpy', 'h5py',
+                           'ndindex', 'pyflakes', 'pytest-cov']):
```
Comment (asmeurer) [inline rever.xsh:11]: This looks like it is missing some dependencies compared to what is installed in Travis.

**Comment by melissawm** [comment]
@scopatz can you take a look when possible so I can do the actual release? Thanks!

**Comment by scopatz** [comment]
Great thanks!

**Pull request merged**